bin/transform

```diff
@@ -257,6 +257,33 @@ def validate_metadata(metadata: pd.DataFrame,
     Performs several tests on the given *metadata*, raising an
     ``AssertionError`` if the tests fail.
     """
+    def annotation_hierarchies(annotations: pd.DataFrame) -> Optional[pd.DataFrame]:
+        """
+        Returns a ``pd.DataFrame` of location hierarchies from the annotations
+        file if annotations are provided, else returns ``None``.
+        """
+        if not args.annotations:
+            return
+
+        annotations.columns = ['strain', 'gisaid_epi_isl', 'variable', 'value']
+        annotations = annotations \
+            .drop_duplicates(['gisaid_epi_isl', 'variable'], keep='last')
+
+        keep_columns = location_hierarchy_columns + \
+            [ f'{resolution}_exposure' for resolution in location_hierarchy_columns]
+        # Avoid a KeyError if the annotations file doesn't contain all of
+        # our resolution columns
+        columns = [ col for col in annotations.columns if col in keep_columns ]
+
+        wide_annotations = annotations \
+            .pivot(index='gisaid_epi_isl', columns='variable', values='value')
+        wide_annotations = wide_annotations[columns].dropna()
+
+        wide_annotations['gisaid_epi_isl'] = wide_annotations.index
+        wide_annotations = wide_annotations.reset_index(drop=True)
```

Comment:
This is kind of clunky IMO. It's some data prep that allows `hierarchy_dataframe()` to run using the expeted `gisaid_epi_isl` column, but it makes me wonder if it's better if `hierarchy_dataframe()` relies on an index instead of a specific, named column for its reshaping.